### PR TITLE
Fix/pr 28/end item timer in pause communication action

### PR DESCRIPTION
**Pull request by SashaTAO**
JIRA ticket: https://oat-sa.atlassian.net/browse/PR-28

Problem:
If test runner is configured to use client timer and messages communication, item timer is not stopped on `pause` action, as it's done in test runner [`pause` endpoint](https://github.com/oat-sa/extension-tao-testqti/blob/develop/actions/class.Runner.php#L814).

How to test: 
- start delivery execution (can be proctored)
- pause delivery execution (test runner plugins or proctor pause)
- check that item timer is stopped. Technical, can be checked in test session's timer, or that event was triggered and delivery monitoring updated with correct value (if proctoring is used).

### CODE REVIEW

File: models/classes/runner/synchronisation/action/Pause.php [hunk 1]
```diff
@@ -65,4 +67,16 @@ public function process()
 
         return $response;
     }
+
+    private function shouldTimerStopOnPause(): bool
+    {
+        $isTerminated = $this->getRunnerService()->isTerminated($this->getServiceContext());
```
Comment (zagovorichev) [inline models/classes/runner/synchronisation/action/Pause.php:73]: too many conditions and it is not so easy to understand when it should be stoped. Maybe it has sense to add extra method?
```
func shouldTimerStopOnPause() :
 return $this->shouldStopByClientsTimer
```

**Review by zagovorichev — CHANGES_REQUESTED** [review]

**Review by jsconan — APPROVED** [review]
The fix looks ok.

**Review Checklist**
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`

File: manifest.php [hunk 2]
```diff
@@ -53,7 +53,7 @@
     'label' => 'QTI test model',
     'description' => 'TAO QTI test implementation',
     'license' => 'GPL-2.0',
-    'version' => '40.2.4',
+    'version' => '40.2.5',
```
[the remark this answers is not in the document]
Comment (jsconan) [reply manifest.php:56]: This is actually a fix. The pause action was improved for the regular request, not for the synchronized actions. This PR aims to fix this discrepancy.

**Review by tikhanovichA — APPROVED** [review]
**Review Checklist**
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`

File: manifest.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (zagovorichev) [reply manifest.php:56]: Yes, for the task, but not for the code.
I thought so: if we add a new behavior model to the code, it is a new functionality, with the addition of new tests for that functionality.

File: models/classes/runner/synchronisation/action/Pause.php (hunk 1 above)
Comment (SashaTAO) [reply models/classes/runner/synchronisation/action/Pause.php:73]: I'm not sure `shouldStopByClientsTimer()` will be useful here. Especially taking into account that it is a temporary fix until proper refactoring by test runner team. We are checking just two conditions, if session was not terminated and timer target is "client". I can refactor code a bit to make it more readable.

**Review by zagovorichev — APPROVED** [review]

**Pull request merged**